internal/streams/streams.go

```diff
@@ -24,7 +24,13 @@ func New(protocol int) *IDGenerator {
 	if protocol > 2 {
 		maxStreams = 32768
 	}
+	return NewLimited(maxStreams)
+}
 
+func NewLimited(maxStreams int) *IDGenerator {
+	if maxStreams < 128 {
```

Comment:
Why? I will definitely want to configure it to less than 128.

Reply:
Sure.